knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java

```diff
@@ -77,17 +77,18 @@ public void doService() {
 			} else if (OPER_DOWNLOAD.equalsIgnoreCase(operation)) {
 				freezeHttpResponse();
 				String fileName = (String) getAttribute("fileName");
+
 				File file = getFile(fileName);
-				try (FileInputStream fis = new FileInputStream(file)) {
+				if (!file.exists()) {
+					writeBackToClient(404, "File not found.", false, null, "text/plain");
+				} else {
+					FileInputStream fis = new FileInputStream(file);
```

Comment:
You removed the try-with-resources block, stream will stay open intead of being closed (every stream must be closed).
Use:
try (FileInputStream fis = new FileInputStream(file)) {
  HttpServletResponse response = getHttpResponse();
  response.setHeader("Content-Disposition", "attachment; filename=\"" + file.getName() + "\";");
  byte[] content = SpagoBIUtilities.getByteArrayFromInputStream(fis);
  response.setContentLength(content.length);
  response.getOutputStream().write(content);
  response.getOutputStream().flush();
}

Reply:
Wrapped the code with try block.